examples/plot_infer_spatial_correlation.py

```diff
@@ -0,0 +1,56 @@
+"""
+Spatial correlation of errors
+=============================
+
+Digital elevation models have errors that are spatially correlated due to instrument or processing effects. Here, we
+rely on a non-stationary spatial statistics framework to estimate and model spatial correlations in elevation error.
+We use a sum of variogram forms to model this correlation, with stable terrain as an error proxy for moving terrain.
+
+**Reference**: `Hugonnet et al. (2022) <https://doi.org/10.1109/jstars.2022.3188922>`_, Figure 5 and Equations 13–16.
+"""
+# sphinx_gallery_thumbnail_number = 1
+import xdem
+import geoutils as gu
+
+# %%
+# We load a difference of DEMs at Longyearbyen, already coregistered using :ref:`coregistration_nuthkaab` as shown in
+# the :ref:`sphx_glr_auto_examples_plot_nuth_kaab.py` example. We also load the glacier outlines here corresponding to
+# moving terrain.
+dh = xdem.DEM(xdem.examples.get_path("longyearbyen_ddem"))
+glacier_outlines = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
+
+# %%
+# Then, we run the pipeline for inference of elevation heteroscedasticity from stable terrain (*Note: we pass a
+# random_state argument to ensure a fixed, reproducible random subsampling in this example*):
+df_empirical_variogram, df_model_params, spatial_corr_function = \
+    xdem.spatialstats.infer_spatial_correlation_from_stable(dvalues=dh, list_models=['Gaussian', 'Spherical'],
+                                                            unstable_mask=glacier_outlines, random_state=42)
+
+# %%
+# The first output corresponds to the dataframe of the empirical variogram, by default estimated using Dowd's estimator
+# and the circular sampling scheme of `skgstat.RasterEquidistantMetricSpace` (Fig. S13 of Hugonnet et al. (2022)):
```

Comment:
Maybe quickly explain the meaning of each column?

Reply:
Done